crates/matrix-sdk/src/error.rs

```diff
@@ -168,6 +169,25 @@ pub enum Error {
     #[cfg(feature = "image-proc")]
     #[error(transparent)]
     ImageError(#[from] ImageError),
+
+    /// An other error was raised and formatted with eyre
+    /// this might happen because encryption was enabled on the base-crate
+    /// but not here and that raised.
+    #[cfg(feature = "eyre")]
+    OtherEyreReport(eyre::Report),
+
+    /// An other error was raised and formatted with anyhow
+    /// this might happen because encryption was enabled on the base-crate
+    /// but not here and that raised.
+    #[cfg(feature = "anyhow")]
+    OtherAnyhowError(anyhow::Error),
+
+    /// An other error was raised and neither eyre, nor anyhow were activated
+    /// so it's Debug information was formatted as a String.
+    /// this might happen because encryption was enabled on the base-crate
+    /// but not here and that raised.
+    #[error("unknown error: {0}")]
+    UnknownError(String),
```

Comment:
Why have these three separate types? When creating `eyre::Report` or `anyhow::Error` directly from an underlying error (no explicit addition of context or such) I don't think it has much advantage over `Box<dyn Error + Send + Sync>`.